federatedscope/core/sampler.py

```diff
@@ -0,0 +1,124 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+
+class Sampler(ABC):
+    """
+    The strategies of sampling clients for each training round
+
+    Arguments:
+        client_state: a dict to manager the state of clients (idle or busy)
+    """
+    def __init__(self, client_num):
+        self.client_state = np.asarray([1] * (client_num + 1))
+        # Set the state of server (index=0) to 'working'
+        self.client_state[0] = 0
```

Comment:
it is very confusing. Server state is maintained in  an array named "client_state"

Reply:
Since 'client_id' is numbered from 1 to M, `index=0` for `client_state` is redundant. Although `client_state[0]` is used for `server_state`, it would not be modified during the training process.
Maybe we can declare it in the comments.